File: packages/sdk-ruby/lib/settler/exceptions.rb
# frozen_string_literal: true

module Settler
  # Base exception for all Settler errors
  class SettlerError < StandardError
    attr_reader :status_code

    def initialize(message, status_code: 0)
      @status_code = status_code
      super(message)
    end
  end

  # Network-related errors
  class NetworkError < SettlerError; end

  # Authentication errors
  class AuthenticationError < SettlerError; end

  # Validation errors
  class ValidationError < SettlerError; end

  # Resource not found errors
  class NotFoundError < SettlerError; end

  # Rate limit exceeded errors
  class RateLimitError < SettlerError; end

  # Server errors
  class ServerError < SettlerError; end
end

File: packages/sdk-python/settler/exceptions.py
"""
Settler Python SDK Exceptions
"""


class SettlerError(Exception):
    """Base exception for all Settler errors"""
    
    def __init__(self, message: str, status_code: int = 0):
        self.message = message
        self.status_code = status_code
        super().__init__(self.message)


class NetworkError(SettlerError):
    """Network-related errors (timeout, connection, etc.)"""
    pass


class AuthenticationError(SettlerError):
    """Authentication errors (invalid API key, expired token, etc.)"""
    pass


class ValidationError(SettlerError):
    """Validation errors (invalid input, missing fields, etc.)"""
    pass


class NotFoundError(SettlerError):
    """Resource not found errors"""
    pass


class RateLimitError(SettlerError):
    """Rate limit exceeded errors"""
    pass


class ServerError(SettlerError):
    """Server errors (5xx)"""
    pass
